daemon/daemon.go

```diff
@@ -259,12 +264,14 @@ func (daemon *Daemon) restore() error {
 				log.Debugf("not restoring container because it was created with another storage driver (%s)", c.Driver)
 				return
 			}
-			rwlayer, err := daemon.imageService.GetLayerByID(c.ID)
-			if err != nil {
-				log.WithError(err).Error("failed to load container mount")
-				return
+			if accessor, ok := daemon.imageService.(layerAccessor); ok {
+				rwlayer, err := accessor.GetLayerByID(c.ID)
+				if err != nil {
+					log.WithError(err).Error("failed to load container mount")
+					return
+				}
+				c.RWLayer = rwlayer
```

Comment:
Discussing this with @rumpl - it's kinda ugly that we're mutating the container that we received from `c, err := daemon.load(id)` here (although this was already the case before this PR)

Wondering if `c, err := daemon.load(id)` should just return the container with the right options already set (or if needed some wrapper).

Reply:
I wonder why don't we use the layer store directly? `GetLayerByID` is literally a one-line wrapper around the layer store 

https://github.com/moby/moby/blob/ad9d70b0e6833b746eecccc08a17459ba8bb31c4/daemon/images/service.go#L140-L142

Reply:
I updated one instance, but (saw you also tried) looks like we're using it elsewhere. I also fixed two comments that were no longer correct, and added a `Warn()` log for an unhandled error when releasing the layer;

https://github.com/moby/moby/compare/14cc4c994e7f653512f95a7b31fc90cc0f09a0ba..1f8b444f1bdd94d8a19116753cf83607d7479029


One thing we should probably look into is that in some cases we call `i.layerstore.ReleaseRWLayer()` directly, and in other cases we call `i.ReleaseLayer()`